packages/stark-ui/src/modules/generic-search/components/generic-search/generic-search.component.ts

```diff
@@ -195,31 +195,31 @@ export class StarkGenericSearchComponent extends AbstractStarkUiComponent implem
 	 * See https://angular.io/guide/template-syntax#template-reference-variables--var- for more info about template reference variables.
 	 */
 	@ContentChild("searchForm")
-	public searchFormComponent: StarkSearchFormComponent<unknown>;
+	public searchFormComponent?: StarkSearchFormComponent<unknown>;
 
 	/**
 	 * Configuration object for the the {@link StarkActionBarComponent} to be shown in the search form.
 	 */
-	public actionBarConfig: StarkActionBarConfig;
+	public actionBarConfig: StarkActionBarConfig = { actions: [] };
 
 	/**
 	 * Object containing normalized options that will be used to generate config for the the {@link StarkActionBarComponent}
 	 * to be shown in the search form.
 	 */
-	public normalizedFormActionBarConfig: StarkGenericSearchActionBarConfigRequired;
+	public normalizedFormActionBarConfig!: StarkGenericSearchActionBarConfigRequired;
 
 	/**
 	 * Object containing normalized options for the different buttons to be shown in the search form.
 	 */
-	public normalizedFormButtonsConfig: StarkGenericSearchFormButtonsConfigRequired;
+	public normalizedFormButtonsConfig!: StarkGenericSearchFormButtonsConfigRequired;
 
 	/**
 	 * Reference to the FormGroup instance bound to the search form.
 	 *
 	 * In the HTML, the component defining the search form content should be exported as `searchForm`
 	 * so that it is accessible from the Stark Generic Search component. See example above.
 	 */
-	public genericForm: FormGroup;
+	public genericForm!: FormGroup;
```

Comment:
I think we should use the `?` modifier here because the `genericForm` depends on the fact that the `searchFormComponent` is present... and that one has the `?` modifier too

Reply:
I refactored this a bit to
```
	public get genericForm(): FormGroup | void {
		if (typeof this.searchFormComponent !== "undefined") {
			return this.searchFormComponent.searchForm;
		}
	}
```
but you are correct about the optional. Also, looking at the code again I think maybe I should mark `searchFormComponent` as definite and add an assert in the onInit.
following the logic of this comment https://github.com/angular/angular/issues/24571#issuecomment-404606595
What do you think?

Reply:
Yes, I agree that `searchFormComponent` should be marked as definite and check it on the `ngOnInit` or the other hook to check child content.

So you can remove the setter you just added ;)